datastore/query.go

```diff
@@ -566,6 +587,8 @@ type Iterator struct {
 	// limit is the limit on the number of results this iterator should return.
 	// A negative value means unlimited.
 	limit int32
+	// count is the number of results this iterator should fetch at once.
```

Comment:
`t.count` can take equal to or greater than zero.  Negative values are rejected when `BatchSize()` sets `q.count`, and `t.count` is set as to be equal to `q.count` when constructing `&Iterator{}`.  So I add note that it can take zero & positive values.

Reply:
I'm not sure 0 makes sense for `count`. In fact, the `if` block we've been discussing checks that it's greater than 0. I feel like `BatchSize` should reject `count` values <= 0.

Reply:
Indeed, that's correct.  I fix `BatchSize`.